merlin/models/tf/metrics/evaluation.py

```diff
@@ -0,0 +1,292 @@
+
+import tensorflow as tf
+from keras.utils.tf_utils import is_tensor_or_variable
+from tensorflow.keras import backend
+from tensorflow.keras.metrics import Mean
+from tensorflow.keras.metrics import get as get_metric
+
+from merlin.models.tf.metrics import metrics_registry
+from merlin.models.tf.utils.tf_utils import get_candidate_probs
+
+MIN_FLOAT = 1e-16
+METRIC_PARAMETERS_DOCSTRING = """
+    predicted_candidates_probs: tf.Tensor
+        A tensor with shape (batch_size, n_items) corresponding to
+        the sorted predicted candidate probabilities. Where the item probability is
+        derived from the frequency of the items in the training set as follow:
+        `item_prob = item_freq_count / sum(item_freq_count)`
+    k : int
+        The cut-off for popularity metrics
+"""
+
+
+def novelty_at(
+    predicted_candidates_probs: tf.Tensor,
+    k: int = 5,
+) -> tf.Tensor:
+    """
+    Computes novely@K metric
+    ----------
+    {METRIC_PARAMETERS_DOCSTRING}
+    """
+    return -tf.math.log(predicted_candidates_probs[:, :k] + MIN_FLOAT)
+
+
+def popularity_bias_at(
+    predicted_candidates_probs: tf.Tensor,
+    k: int = 5,
+) -> tf.Tensor:
+    """
+    Computes popularity bias@K metric
+    ----------
+    {METRIC_PARAMETERS_DOCSTRING}
+    """
+    return tf.reduce_sum(predicted_candidates_probs[:, :k], -1) / k
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class PopularityMetric(Mean):
+    """
+    Parameters
+    ----------
+    fn:
+        The popularity metric function to wrap,
+        with signature `fn(predicted_candidates_probs, k, **kwargs)`.
+
+    item_freq_probs: Union[tf.Tensor, Sequence]
+        A Tensor or list with item frequencies (if is_prob_distribution=False)
+        or with item probabilities (if is_prob_distribution=True)
+    is_prob_distribution: bool, optional
+        If True, the item_freq_probs should be a probability distribution of the items.
+        If False, the item frequencies is converted to probabilities, by default True
+    k: int, optional
+        The cut-off for popularity metrics, by default 5
+    """
+
+    def __init__(
+        self,
+        fn,
+        item_freq_probs: Union[tf.Tensor, Sequence],
+        k=5,
+        is_prob_distribution: bool = True,
+        name: str = None,
+        dtype=None,
+        **kwargs,
+    ):
+        if name is not None:
+            name = f"{name}_{k}"
+        super().__init__(name=name, dtype=dtype)
+        self._fn = fn
+        self.k = k
+        self._fn_kwargs = kwargs
+        self.is_prob_distribution = is_prob_distribution
+
+        candidate_probs = get_candidate_probs(item_freq_probs, self.is_prob_distribution)
+        self.candidate_probs = tf.Variable(
+            candidate_probs,
+            name="candidate_probs",
+            trainable=False,
+            dtype=tf.float32,
+            validate_shape=False,
+            shape=tf.shape(candidate_probs),
+        )
```

Comment:
I think we could have a method `PopularityMetric.update_candidate_probs()` like [here](https://github.com/NVIDIA-Merlin/models/blob/main/merlin/models/tf/blocks/core/transformations.py#L673), to allow for users being able to update the probs.
That could be useful for example if training on streaming data, where the items popularity could be adjusted over time